lib/nylas/http_client.rb

```diff
@@ -104,8 +104,17 @@ def execute(method:, path: nil, headers: {}, query: {}, payload: nil)
                         also_log: { result: true, values: %i[method url path headers query payload] }
 
     def build_request(method:, path: nil, headers: {}, query: {}, payload: nil, timeout: nil)
-      headers[:params] = query
       url ||= url_for_path(path)
+
+      # Add parameters using the uri module as it allows us to make custom rules
+      unless query.empty?
```

Comment:
We need to add some tests for this to make sure we are not making anything wrong here.